### PR TITLE
fix : Content: No Fields Have Been Added Screen appears for Content Items that have fields

### PR DESCRIPTION
**Pull request by glespinosa**
Removed it by adding condition if the query is done that's the only time the actionFields.length will be use

will close https://github.com/zesty-io/manager-ui/issues/2506

### CODE REVIEW

**Comment by glespinosa** [comment]
> @glespinosa not sure if this solves the issue. How did you reproduce the issue?

**Cause of issue**: actionFields.length returns [] empty array on first render of the page which returns 0 value of its length thats why it's partially showing the pic below
image

**Solution**: to resolve this issue I added another checking if the query is already done and ready to read that's the reason for isSuccess and isFetching


To reproduce this try removing the isSuccess and isFetching then go to content with create item it will show this label
"No fields have been added"
image

**Comment by glespinosa** [comment]
> @glespinosa this solution will cause the fields not to render and then just pop in after a while that is still not a great experience especially when there is a loading screen that shows up prior to it. I would expect when the loader disappears then everything for the view is loaded.
> 
> Screenshot 2024-01-31 at 5 02 32 PM


@agalin920  I think to achieve that I added extra code.. I updated the ItemCreate.tsx, I run the useGetContentModelFieldsQuery to get the isSuccess,isFetching that will be added on the condition for the WithLoader component to ensure that all fields will be rendered after the loading screen. While on the Editor component I make sure that if the component being use is for NEW ITEM or ITEM with Existing values, that will determine the value what the activeFields will use.

[screencast-8-aaeffee09b-7w6v22.manager.dev.zesty.io_8080-2024.02.01-14_04_10.webm](https://github.com/zesty-io/manager-ui/assets/44116036/1cacbef1-3490-4b43-9fbc-b673b9fc169c)

File: src/apps/content-editor/src/app/components/Editor/Editor.js [hunk 1]
```diff
@@ -38,18 +38,20 @@ export default memo(function Editor({
   saveClicked,
   onUpdateFieldErrors,
   fieldErrors,
+  newModelFields,
 }) {
   const dispatch = useDispatch();
   const isNewItem = itemZUID.slice(0, 3) === "new";
   const { data: fields } = useGetContentModelFieldsQuery(modelZUID);
+  const currentFields = isNewItem ? newModelFields : fields;
```
[the remark this answers is not in the document]
Comment (glespinosa) [reply src/apps/content-editor/src/app/components/Editor/Editor.js]: removing it now. Pls see updated commits

File: src/apps/content-editor/src/app/views/ItemCreate/ItemCreate.tsx [hunk 2]
```diff
@@ -268,7 +274,12 @@ export const ItemCreate = () => {
   }
 
   return (
-    <WithLoader condition={!loading && item} message="Creating New Item">
+    <WithLoader
+      condition={
+        !loading && item && isSuccessNewModelFields && !isFetchingNewModelFields
```
Comment (theofficialnar) [inline src/apps/content-editor/src/app/views/ItemCreate/ItemCreate.tsx:279]: My only concern here is wouldn't this just keep on showing the loading screen if the api call to get the model fields fail - `isSuccessNewModelFields`?

**Pull request merged**